attestation/verifier-risczero/methods/guest/src/main.rs

```diff
@@ -390,6 +421,53 @@ fn main() {
     let signature = Signature::from_scalars(r, s).unwrap();
 
     verifying_key.verify_prehash(&hash, &signature).unwrap();
+}
 
-    println!("Done!");
+#[cfg(test)]
+mod tests {
+    use std::cell::RefCell;
+    use std::rc::Rc;
+
+    use crate::verify;
+
+    // TODO: add more extensive tests
+
+    // NOTE: Seems a bit convoluted, idk if it can be simplified
+    fn create_committer() -> (Rc<RefCell<Vec<u8>>>, impl Fn(&[u8])) {
+        let env = Rc::new(RefCell::new(vec![]));
+
+        let env_clone = env.clone();
+
+        let commit_slice = move |slice: &[u8]| {
+            env_clone.borrow_mut().extend_from_slice(slice);
+        };
+
+        (env, commit_slice)
+    }
+
+    #[test]
+    fn test_aws() {
+        let attestation =
+            std::fs::read(file!().rsplit_once('/').unwrap().0.to_owned() + "/testcases/aws.bin")
+                .unwrap();
+
+        let (journal, committer) = create_committer();
+
+        verify(&attestation, committer);
+
+        assert_eq!("00000193bef3f3b0189038eccf28a3a098949e402f3b3d86a876f4915c5b02d546abb5d8c507ceb1755b8192d8cfca66e8f226160ca4c7a65d3938eb05288e20a981038b1861062ff4174884968a39aee5982b312894e60561883576cc7381d1a7d05b809936bd166c3ef363c488a9a86faa63a44653fd806e645d4540b40540876f3b811fc1bceecf036a4703f07587c501ee45bb56a1aafc0254eba608c1f36870e29ada90be46383292736e894bfff672d989444b5051e534a4b1f6dbe3c0bc581a32b7b176070ede12d69a3fea211b66e752cf7dd1dd095f6f1370f4170843d9dc100121e4cf63012809664487c9796284304dc53ff440e646f8b0071d5ba75931402522cc6a5c42a84a6fea238864e5ac9a0e12d83bd36d0c8109d3ca2b699fce8d082bf313f5d2ae249bb275b6b6e91e0fcd9262f4bb0000", hex::encode(journal.borrow_mut().as_slice()));
```

Comment:
we can put the expected result in a variable and use that to compare.

Reply:
as in, have fields and encode in the test?

Reply:
```suggestion
        let expected_journal = "00000193bef3f3b0189038eccf28a3a098949e402f3b3d86a876f4915c5b02d546abb5d8c507ceb1755b8192d8cfca66e8f226160ca4c7a65d3938eb05288e20a981038b1861062ff4174884968a39aee5982b312894e60561883576cc7381d1a7d05b809936bd166c3ef363c488a9a86faa63a44653fd806e645d4540b40540876f3b811fc1bceecf036a4703f07587c501ee45bb56a1aafc0254eba608c1f36870e29ada90be46383292736e894bfff672d989444b5051e534a4b1f6dbe3c0bc581a32b7b176070ede12d69a3fea211b66e752cf7dd1dd095f6f1370f4170843d9dc100121e4cf63012809664487c9796284304dc53ff440e646f8b0071d5ba75931402522cc6a5c42a84a6fea238864e5ac9a0e12d83bd36d0c8109d3ca2b699fce8d082bf313f5d2ae249bb275b6b6e91e0fcd9262f4bb0000";
        assert_eq!(expected_journal, hex::encode(journal.borrow_mut().as_slice()));
```
like this. It was just a nit comment.